presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometrySerializationType.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import static java.util.Objects.requireNonNull;
+
+public enum GeometrySerializationType
```

Comment:
If we didn't have two different `*Serde` i would move this class inside the `*Serde` and made it private. Despite it can't be done - i would keep the semantic to be the same. Nothing outside the `Serde` classes shouldn't use this guy directly. Since `SerializationType` is something strongly related to `Serialization`. What i would **suggest** you to do here is

1. Add `GeometrySerde#getShapeType(Slice)` method, which internally will deserialize only the serialization type and map it to geometry type.
2. Replace all the `GeometrySerializationType` usages with the method above.
3. Create a separate package, called `com.facebook.presto.geospatial.serde`, move all related to serde stuff there, including serde tests. 
4. Make `GeometrySerializationType` to be package private

Feel free to squash all your commits afterwards, but leave only two.

- refactor
- testSTGeometryType implementation